src/EnergyPlus/PluginManager.cc

```diff
@@ -122,13 +129,16 @@ void onBeginEnvironment(EnergyPlusData &state)
 
 int PluginManager::numActiveCallbacks(EnergyPlusData &state)
 {
-    return (int)state.dataPluginManager->callbacks.size();
+    return (int)state.dataPluginManager->callbacks.size() + (int)state.dataPluginManager->userDefinedCallbacks.size();
 }
 
 void runAnyRegisteredCallbacks(EnergyPlusData &state, EMSManager::EMSCallFrom iCalledFrom, bool &anyRan)
 {
     if (state.dataGlobal->KickOffSimulation) return;
     for (auto const &cb : state.dataPluginManager->callbacks[iCalledFrom]) {
+        if (iCalledFrom == EMSManager::EMSCallFrom::UserDefinedComponentModel) {
+            continue; // these are called -intentionally- using the runSingleUserDefinedCallback method
+        }
```

Comment:
User-defined callbacks are intended to be called intentionally by name.  This extra check may not be necessary, but I definitely wouldn't want to accidentally call one at the wrong time, so it seems like a tiny safety addition.